src/client.rs

```diff
@@ -77,6 +77,9 @@ impl<S: AsyncRead + AsyncWrite + Unpin + Send> Client<S> {
     /// [`ToSql`]. For reading data from the database, see the documentation for
     /// [`FromSql`].
     ///
+    /// This API is not quite suitable for dynamic query parameters. In these
+    /// cases using a [`Query`] object might be easier.
```

Comment:
We could expand here and say why: the query parameters need to have the same lifetime, you can't pass them around by value (in my understanding).